### PR TITLE
test: rewrite tests

### LINKED ISSUE #11: About testing
Continue https://github.com/alexjoverm/tslint-config-prettier/issues/8#issuecomment-322700128.

I'd prefer using my structure ([unit tests](https://github.com/ikatyang/tslint-config-prettier-ext/tree/master/rules/tslint) using [some transformer](https://github.com/ikatyang/tslint-config-prettier-ext/blob/master/src/transform/process.ts) to generate tests), what do you think?

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #11 

- [x] Let's keep src folder for the rulesets, and create a tools folder for the runtime and so on
- [x] Please use camelCase instead of sneak_case (including the preprocess)
- I'd add a `expect(errorCountAfter).toBe(0)` line
  - https://github.com/alexjoverm/tslint-config-prettier/issues/11#issuecomment-325189988

---

Things I've done:

- [x] pin tslint/tslint-*/prettier/typescript version

---

Things I want to do, but I don't know if I can:

- [ ] add yarn.lock
  - [ ] setup [greenkeeper-lockfile](https://github.com/greenkeeperio/greenkeeper-lockfile)
- [ ] add a *Related* section on `readme.md` linked to [tslint-plugin-prettier](https://github.com/ikatyang/tslint-plugin-prettier)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,9 +24,12 @@
...
-    "test:lint": "tslint 'test-lint/**'",
-    "test:prod": "npm t && npm run test:lint",
+    "prepublish": "npm run build-tools && npm run generate-config",
+    "lint": "tslint 'tools/**'",
+    "test": "npm run clean && jest --no-cache",
+    "clean": "rimraf ./node_modules/vrsource-tslint-rules/rules/*.ts",
+    "build-tools": "tsc",
+    "generate-config": "node ./build/generate.js",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post",
     "semantic-release-prepare": "node tools/semantic-release-prepare",
```
Comment (alexjoverm) [inline package.json]: Seems like this is not used

File: package.json [hunk 2]
```diff
@@ -35,9 +24,12 @@
     "node": ">=4.0.0"
   },
   "scripts": {
-    "test": "jest",
-    "test:lint": "tslint 'test-lint/**'",
-    "test:prod": "npm t && npm run test:lint",
+    "prepublish": "npm run build-tools && npm run generate-config",
+    "lint": "tslint 'tools/**'",
+    "test": "npm run clean && jest --no-cache",
+    "clean": "rimraf ./node_modules/vrsource-tslint-rules/rules/*.ts",
```
Comment (alexjoverm) [inline package.json]: What is this for? Why do we need to remove something inside node_modules?

Comment (ikatyang) [reply package.json]: It's an tslint issue, tslint can't load rules that their *.ts and *.js exist simultaneously (while using `rulesDirectory`), unfortunately, `vrsource-tslint-rules` is the only ruleset that publish *.ts with *.js.

image

**Comment by ikatyang** [comment]
> What about renaming all rules.js to rules.json. Since we're using typescript, and rules can be just a json files, then we don't keep all js, json and ts along the repo

Sorry for the confusing, I forgot to remove those `rules.js` since we currently generate `rules.json `dynamically, you can found the generated `rules.json` in the local `lib` folder, it's gitignored so it does not display on github.

> About your questions: yes, you can do both ;)

I'll add them in another PR, thanks. :)

File: package.json (hunk 2 above)
Comment (alexjoverm) [reply package.json]: Interesting :D good catch

**Comment by alexjoverm** [comment]
Thanks @ikatyang for your great work and for your patience, we finally got it :)

**Pull request merged**